coderd/workspacebuilds.go

```diff
@@ -278,28 +278,64 @@ func (api *API) postWorkspaceBuilds(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// we only want to create audit logs for delete builds right now
+	auditor := api.Auditor.Load()
+
+	// if user deletes a workspace, audit the workspace
 	if action == rbac.ActionDelete {
 		var (
-			auditor           = api.Auditor.Load()
 			aReq, commitAudit = audit.InitRequest[database.Workspace](rw, &audit.RequestParams{
-				Audit:   *auditor,
-				Log:     api.Logger,
-				Request: r,
-				Action:  database.AuditActionDelete,
+				Audit:            *auditor,
+				Log:              api.Logger,
+				Request:          r,
+				Action:           database.AuditActionDelete,
+				AdditionalFields: json.RawMessage("{}"),
 			})
 		)
 
 		defer commitAudit()
 		aReq.Old = workspace
 	}
 
+	latestBuild, latestBuildErr := api.Database.GetLatestWorkspaceBuildByWorkspaceID(ctx, workspace.ID)
+
+	// if a user starts/stops a workspace, audit the workspace build
+	if action == rbac.ActionUpdate {
+		var auditAction database.AuditAction
+		if createBuild.Transition == codersdk.WorkspaceTransitionStart {
+			auditAction = database.AuditActionStart
+		} else if createBuild.Transition == codersdk.WorkspaceTransitionStop {
+			auditAction = database.AuditActionStop
+		} else {
+			auditAction = database.AuditActionWrite
+		}
+
+		// We pass the workspace name to the Auditor so that it
+		// can form a friendly string for the user.
+		workspaceResourceInfo := map[string]string{
+			"workspaceName": workspace.Name,
+		}
+
+		wriBytes, _ := json.Marshal(workspaceResourceInfo)
+
+		var (
+			aReq, commitAudit = audit.InitRequest[database.WorkspaceBuild](rw, &audit.RequestParams{
+				Audit:            *auditor,
+				Log:              api.Logger,
+				Request:          r,
+				Action:           auditAction,
+				AdditionalFields: wriBytes,
+			})
+		)
```

Comment:
small nit: since this is just a single variable you don't really need the `var ()` block, you can just use `:=`.